src/promptflow/promptflow/_sdk/data/executable/utils.py

```diff
@@ -0,0 +1,131 @@
+                result = True
+    return result
+
+
+def is_dict_contains_rich_text(rich_text):
+    result = False
+    for rich_text_key, rich_text_value in rich_text.items():
+        if isinstance(rich_text_value, list):
+            result |= is_list_contains_rich_text(rich_text_value)
+        elif isinstance(rich_text_value, dict):
+            result |= is_dict_contains_rich_text(rich_text_value)
+        elif re.match(MIME_PATTERN, rich_text_key) or (isinstance(rich_text_value, str) and
+                                                       rich_text_value.startswith("data:image")):
+            result = True
+    return result
+
+
+def item_render_message(value, key=None):
+    if key and re.match(MIME_PATTERN, key):
+        show_image(value, key)
+    elif isinstance(value, str) and value.startswith("data:image"):
+        show_image(value)
+    else:
+        if key is None:
+            st.markdown(f"`{json_dumps(value)},`")
+        else:
+            st.markdown(f"`{key}: {json_dumps(value)},`")
+
+
+def list_iter_render_message(message_items):
+    if is_list_contains_rich_text(message_items):
+        st.markdown("`[ `")
+        for item in message_items:
+            if isinstance(item, list):
+                list_iter_render_message(item)
+            if isinstance(item, dict):
+                dict_iter_render_message(item)
+            else:
+                item_render_message(item)
+        st.markdown("`], `")
+    else:
+        st.markdown(f"`{json_dumps(message_items)},`")
+
+
+def dict_iter_render_message(message_items):
+    if is_multimedia_dict(message_items):
+        key = list(message_items.keys())[0]
+        value = message_items[key]
+        show_image(value, key)
+    elif is_dict_contains_rich_text(message_items):
+        st.markdown("`{ `")
+        for key, value in message_items.items():
+            if re.match(MIME_PATTERN, key):
+                show_image(value, key)
+            else:
+                if isinstance(value, list):
+                    st.markdown(f"`{key}: `")
+                    list_iter_render_message(value)
+                elif isinstance(value, dict):
+                    st.markdown(f"`{key}: `")
+                    dict_iter_render_message(value)
+                else:
+                    item_render_message(value, key)
+        st.markdown("`}, `")
+    else:
+        st.markdown(f"`{json_dumps(message_items)},`")
+
+
+def extract_content(node):
+    if isinstance(node, NavigableString):
+        text = node.strip()
+        if text:
+            return [text]
+    elif isinstance(node, Tag):
+        if node.name == 'img':
+            prefix, base64_str = node['src'].split(',', 1)
+            return [{prefix: base64_str}]
+        else:
+            result = []
+            for child in node.contents:
+                result.extend(extract_content(child))
+            return result
+    return []
+
+
+def parse_html_content(html_content):
```

Comment:
may change the function name here: parse_list_from_html

add some docstring on that it will extract all <p>